lib/ecto/changeset.ex

```diff
@@ -729,6 +738,9 @@ defmodule Ecto.Changeset do
         case Map.get(types, key) do
           {:embed, embed} ->
             {key, Ecto.Embedded.apply_changes(embed, value)}
+          {:assoc, _assoc} ->
+            raise ArgumentError, "associations should be removed from changeset" <>
```

Comment:
We are using `apply_changes` to build a preview before saving into the DB. Right now, with no nested changesets support we have to manually propagate application in order to build this preview. It would be sad to keep this mess even when Ecto will be able to handle this for saving.